odbc/src/api/data.rs

```diff
@@ -1299,7 +1295,7 @@ pub mod isize_len {
 /// # Safety
 /// This writes to a raw C-pointers
 ///
-pub unsafe fn set_str_length(string_length_ptr: *mut Integer, length: Integer) {
+pub unsafe fn set_str_length<T>(string_length_ptr: *mut T, length: T) {
```

Comment:
Should we change the name of this function since it's now used more generally? maybe like `set_data_length`?